node/packages/parser/src/ethereum/yearn.ts

```diff
@@ -0,0 +1,33 @@
+import { Tx } from '@shapeshiftoss/blockbook'
+import { TxSpecific as ParseTxSpecific, YearnTx } from '../types'
+import shapeShiftRouter from './abi/shapeShiftRouter'
+import yearnVault from './abi/yearnVault'
+import { addABI, decodeMethod } from 'abi-decoder'
+import { GenericParser } from './index'
+
+const ROUTER_CONTRACT = '0x6a1e73f12018D8e5f966ce794aa2921941feB17E'
+
+export class Parser implements GenericParser {
+  supportsTransaction(tx: Tx) {
+    const data = tx.ethereumSpecific?.data
+    const firstVoutAddress = tx.vout && tx.vout[0].addresses && tx.vout[0].addresses[0]
+    const contractData = data?.slice(0, 2).concat(data?.slice(34, 74)).toLowerCase()
+
+    if (firstVoutAddress === ROUTER_CONTRACT) return true
+    if (contractData === ROUTER_CONTRACT.toLowerCase()) return true
+
+    return false
+  }
+
+  async parse(tx: Tx): Promise<ParseTxSpecific<YearnTx> | undefined> {
+    if (!this.supportsTransaction(tx)) return
+    addABI([...shapeShiftRouter, ...yearnVault])
```

Comment:
How does decoding resolve if there are matching functions in each abi? Not sure if that is the case, but with loading both and decoding, it seems like a scenario we would want to understand. 

This would be another case in my opinion in which by knowing which contract we are interacting with, we could explicitly decode the input data against the specific abi.

Reply:
You're right, it doesn't handle it well.

In the updated `ethers` implementation, I've added an ABI detection switch so we explicitly specify which one we're using.

https://github.com/shapeshift/unchained/pull/267/commits/2953d65c9a978276a520a2f9707dcb3c1381583e